### ISSUE TITLE
Something is left behind in uninstallation, breaking some future deployments

### ISSUE THREAD

**Issue by Sarten-X** [signalfx/splunk-otel-collector-chart#354]
I deployed the Helm chart to a cluster, and all is fine. Data appears in Splunk and I can do charts and dashboards and all the usual things. However, I had deployed it to the `default` namespace, and it really shouldn't be there... So I uninstalled the chart, and deployed it again, this time to a `splunk `namespace.

Now almost everything is working again, except the cluster receiver pod. That one is giving a "permission denied" error when the pod tries to start. I can force the pod to start if I give the service account full cluster-admin permission, but that's bad.
We have a test cluster. When I deploy there, specifying the `splunk` namespace the first time, all the pods work. Then if I uninstall and deploy to `default` namespace, that same pod shows the same error.

If the chart is uninstalled again and redeployed to the original namespace again, the cluster receiver works again. 

Something permission-related is not getting cleared during the uninstallation, but I don't know what.

**Comment by dmitryax** [comment]
Hi @Sarten-X,

The helm chart is supposed to be installed in the namespace that is currently set as default for a user running the helm command, that is expected behavior. The namespace can be changed with `-n, --namespace` helm argument. We don't document it to avoid duplicating [official helm docs](https://helm.sh/docs/helm/helm_install/), but maybe we should add a note about the namespace.

I tried to reproduce the issue of potential resources that are not cleaned up, but no luck, it works as expected for me. I did the following:

```
➜  helm uninstall my-splunk-otel-collector
release "my-splunk-otel-collector" uninstalled

➜  helm install --namespace splunk --create-namespace my-splunk-otel-collector -f myaws.yaml splunk-otel-collector-chart/splunk-otel-collector
NAME: my-splunk-otel-collector
LAST DEPLOYED: Mon Jan 17 09:33:58 2022
NAMESPACE: splunk
STATUS: deployed
REVISION: 1
TEST SUITE: None
NOTES:
Splunk OpenTelemetry Collector is installed and configured to send data to Splunk Observability realm us0.

➜ kgp -n splunk
NAME                                                            READY   STATUS    RESTARTS   AGE
my-splunk-otel-collector-agent-97jh2                            1/1     Running   0          23s
my-splunk-otel-collector-agent-dk6js                            1/1     Running   0          23s
my-splunk-otel-collector-agent-snc6t                            1/1     Running   0          23s
my-splunk-otel-collector-k8s-cluster-receiver-7d7759f47-8mkqw   1/1     Running   0          23s
```

Please let me know what commands you used, so I can reproduce it.

**Comment by Sarten-X** [comment]
The (retyped) commands I used were:

`helm install my-splunk-otel-collector -f otelvalues.yaml splunk-otel-collector-chart/splunk-otel-collector`
(time passes)
`helm uninstall my-splunk-otel-collector`
`helm install --namespace splunk --create-namespace my-splunk-otel-collector -f otelvalues.yaml splunk-otel-collector-chart/splunk-otel-collector`

This did not work, as described. I tried it again, and it did not work. I gave up, left it alone for over a month, and just recently got a chance to try installing to the `splunk` namespace again... and it works perfectly fine.

Assuming I'm not hallucinating, I would guess that it was some resource that would eventually be cleaned up after deletion, but for whatever reason Kubernetes didn't clean as quickly as my installation script ran.

Regardless of the cause, I think it's safe to consider this a non-issue unless someone else encounters it. At worst it seems to be something that resolves itself.
